apps/examples/binary_update/binary_update_main.c

```diff
@@ -288,6 +269,46 @@ static void binary_update_run_tests(int repetition_num)
 
 	binary_update_getinfo_all();
 
+#else
+
+	/* get inactive partition path */
+	binary_update_getinfo(APP_NAME, &pre_bin_info);
+
+	/* malloc for arguments */
+	arguments[0] = (char *) malloc ((20 + 1) * sizeof(char));
+	arguments[1] = (char *) malloc ((sizeof(pre_bin_info.active_dev) + 1) * sizeof(char));
+	arguments[2] = (char *) malloc ((15 + 1) * sizeof(char));
+	arguments[3] = (char *) malloc ((sizeof(pre_bin_info.inactive_dev) + 1) * sizeof(char));
+
+	/* Copy path names to arguments */
+	memcpy(arguments[0], "blockwise_bin_update", 20);
+	arguments[0][20] = '\0';
+
+	memcpy(arguments[1], "/dev/mtdblock2", sizeof(pre_bin_info.active_dev));
+	arguments[1][sizeof(pre_bin_info.active_dev)] = '\0';
+
+	memcpy(arguments[2], "/mnt/patch_file", 15);
+	arguments[2][15] = '\0';
+
+	memcpy(arguments[3], "/dev/mtdblock3", sizeof(pre_bin_info.inactive_dev));
+	arguments[3][sizeof(pre_bin_info.inactive_dev)] = '\0';
+
+	printf("arguments[0] = %s\n", arguments[0]);
+	printf("arguments[1] = %s\n", arguments[1]);
+	printf("arguments[2] = %s\n", arguments[2]);
+	printf("arguments[3] = %s\n", arguments[3]);
+
+	/* Patch to a new file */
+	blockwise_bin_update_main(4, arguments);
+
+	/* Reload micom binary from patched partition */
+	binary_update_reload(APP_NAME);
+
+	/* Get info for all */
+	binary_update_getinfo_all();
+
+#endif
```

Comment:
I don't understand why it is different between delta update and full update in application side.